File: api/src/main/java/com/alten/remotesync/application/report/record/request/PagedReportSearchDTO.java
package com.alten.remotesync.application.report.record.request;

import com.alten.remotesync.domain.report.enumeration.ReportStatus;

import java.time.LocalDate;

public record PagedReportSearchDTO(
        Integer pageNumber,
        Integer pageSize,
        String title,
        ReportStatus status,
        LocalDate startDate
) {
}

File: app/src/app/dto/rc/paged-report-search.dto.ts
import { ReportStatus } from '../../enums/report-status.enum';

export interface PagedReportSearchDTO {
  pageNumber?: number;
  pageSize?: number;
  title?: string;
  status?: ReportStatus;
  startDate?: string;
  endDate?: string;
  clientId?: string;
  factoryId?: string;
}
